stratum/hal/lib/barefoot/bf_chassis_manager.cc

```diff
@@ -351,13 +352,87 @@ ::util::Status BFChassisManager::PushChassisConfig(
 
 ::util::Status BFChassisManager::VerifyChassisConfig(
     const ChassisConfig& config) {
-  if (config.trunk_ports_size()) {
-    return MAKE_ERROR(ERR_INVALID_PARAM)
-           << "Trunk ports are not supported on Tofino.";
+  CHECK_RETURN_IF_FALSE(config.trunk_ports_size() == 0)
+      << "Trunk ports are not supported on Tofino.";
+  CHECK_RETURN_IF_FALSE(config.port_groups_size() == 0)
+      << "Port groups are not supported on Tofino.";
+  CHECK_RETURN_IF_FALSE(config.nodes_size() > 0)
+      << "The config must contain at least one node.";
+
+  // Find the supported Tofino chip types based on the given platform.
+  CHECK_RETURN_IF_FALSE(config.has_chassis() && config.chassis().platform())
+      << "Config needs a Chassis message with correct platform.";
+  CHECK_RETURN_IF_FALSE(config.chassis().platform() == PLT_BAREFOOT_TOFINO)
+      << "Unsupported platform: " << Platform_Name(config.chassis().platform());
```

Comment:
I'm surprised this was a) not checked, b) never tripped anyone up.

Reply:
I believed that we also need to add `PLT_BAREFOOT_TOFINO2`

Reply:
whoops